modules/porous_flow/include/materials/PorousFlowPermeabilityKozenyCarman.h

```diff
@@ -0,0 +1,79 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+
+#ifndef POROUSFLOWPERMEABILITYKOZENYCARMAN_H
+#define POROUSFLOWPERMEABILITYKOZENYCARMAN_H
+
+#include "DerivativeMaterialInterface.h"
+#include "Material.h"
+#include "PorousFlowPermeabilityUnity.h"
+
+#include "PorousFlowDictator.h"
+
+//Forward Declarations
+class PorousFlowPermeabilityKozenyCarman;
+
+template<>
+InputParameters validParams<PorousFlowPermeabilityKozenyCarman>();
+
+/**
+ * Material designed to provide the permeability tensor which is calculated 
+ * from porosity using a form of the Kozeny-Carman equation (e.g. Oelkers 
+ * 1996: Reviews in Mineralogy v. 34, p. 131-192):
+ * k = k_ijk * A * phi^n / (1 - phi)^m
+ * where k_ijk is a tensor providing the anisotropy, phi is porosity, 
+ * n and m are positive scalar constants and A is given in one of the
+ * following forms:
+ * A = k0 * (1 - phi0)^m / phi0^n
+ * where k0 and phi0 are a reference permeability and porosity,
+ * or
+ * A = f * d^2
+ * where f is a scalar constant and d is grain diameter.
+ */
+class PorousFlowPermeabilityKozenyCarman : public PorousFlowPermeabilityUnity
+{
+public:
+  PorousFlowPermeabilityKozenyCarman(const InputParameters & parameters);
+
+protected:
+  void computeQpProperties();
+
+  /// Reference scalar permeability in A = k0 * (1 - phi0)^m / phi0^n
+  const Real _k0;
+  
+  /// Reference porosity in A = k0 * (1 - phi0)^m / phi0^n
+  const Real _phi0;
+  
+  /// Multiplying factor in A = f * d^2
+  const Real _f;
+  
+  /// Grain diameter in A = f * d^2
+  const Real _d;
+  
+  /// Exponent in k = k_ijk * A * phi^n / (1 - phi)^m
+  const Real _m;
+
+  /// Exponent in k = k_ijk * A * phi^n / (1 - phi)^m
+  const Real _n;  
+
+  /// Tensor multiplier k_ijk in k = k_ijk * A * phi^n / (1 - phi)^m
+  const RealTensorValue _k_anisotropy;
+
+  /// quadpoint porosity
+  const MaterialProperty<Real> & _porosity_qp;
+  
+  /// d(quadpoint porosity)/d(PorousFlow variable)
+  const MaterialProperty<std::vector<Real> > & _dporosity_qp_dvar;
+
+  /// Name of porosity-permeability relationship
+  const MooseEnum _poroperm_function;
+
+  /// Multiplying factor in k = k_ijk * A * phi^n / (1 - phi)^m
+  Real _A;
```

Comment:
_A is currently initialised in the body of the constructor so I can’t make it const. I would have to put the switch statement into the initialisation list. Is that what I should do?

From: Andy Wilkins [mailto:notifications@github.com]
Sent: Friday, 5 August 2016 8:11 AM
To: WilkAndy/moose <moose@noreply.github.com>
Cc: Sheldon, Heather (Mineral Resources, Kensington WA) <Heather.Sheldon@csiro.au>; Author <author@noreply.github.com>
Subject: Re: [WilkAndy/moose] Initial porous flow 6845 - poroperm functions (#45)


In modules/porous_flow/include/materials/PorousFlowPermeabilityKozenyCarman.h<https://github.com/WilkAndy/moose/pull/45#discussion_r73624453>:

> +  const Real _n;

> +

> +  /// Tensor multiplier k_ijk in k = k_ijk * A * phi^n / (1 - phi)^m

> +  const RealTensorValue _k_anisotropy;

> +

> +  /// quadpoint porosity

> +  const MaterialProperty<Real> & _porosity_qp;

> +

> +  /// d(quadpoint porosity)/d(PorousFlow variable)

> +  const MaterialProperty<std::vector<Real> > & _dporosity_qp_dvar;

> +

> +  /// Name of porosity-permeability relationship

> +  const MooseEnum _poroperm_function;

> +

> +  /// Multiplying factor in k = k_ijk * A * phi^n / (1 - phi)^m

> +  Real _A;

I think you can (should) make this const Real _A, right?

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://github.com/WilkAndy/moose/pull/45/files/ce366d17c08b92cdb7d58f1273857ae5c8851b49#r73624453>, or mute the thread<https://github.com/notifications/unsubscribe-auth/AKJ_8mz6EM9KyCip1pkoEfZJV0dzbjKhks5qcn97gaJpZM4JZPBg>.